dining/views.py

```diff
@@ -0,0 +1,348 @@
+
+        for transaction in transactions:
+            card["data"].append(
+                {"date": transaction.date.isoformat(), "balance": transaction.balance}
+            )
+
+        card["data"] = sorted(card["data"], key=lambda k: k["date"])
+
+        return card
+
+    def get_prediction_swipes(self, uid, start, end):
+
+        balances = DiningBalance.objects.filter(account_id=uid, created_at__gte=start,).order_by(
+            "-created_at"
+        )
+
+        card = {
+            "type": "predictions-graph-swipes",
+            "start-of-semester": start.isoformat(),
+            "end-of-semester": end.isoformat(),
+            "data": [],
+        }
+
+        balance = balances[0].swipes
+
+        if balance != 0:
+
+            swipe_balances = []
+
+            for balance in balances:
+                swipe_balances.append(balance.swipes)
+
+            spent = 0
+
+            for i, balance in enumerate(swipe_balances):
+                if i == 0:
+                    continue
+                else:
+                    if swipe_balances[i] < swipe_balances[i - 1]:
+                        continue
+                    else:
+                        spent += swipe_balances[i] - swipe_balances[i - 1]
+
+            eastern = timezone("US/Eastern")
+            now = datetime.now().replace(tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0)
+            days_since_start = (now - start).days
+            spend_rate = spent / days_since_start
+
+            days_left = int(balance / spend_rate)
+
+            broke_day = now + timedelta(days=days_left)
+
+            card["predicted-zero-date"] = broke_day.isoformat()
+
+            if broke_day > end:
+                days_to_end = (end - now).days
+                card["balance-at-semester-end"] = round(spend_rate * days_to_end, 0)
+
+        for balance in balances:
+            card["data"].append({"date": balance.created_at.isoformat(), "balance": balance.swipes})
+
+        card["data"] = sorted(card["data"], key=lambda k: k["date"])
+
+        return card
+
+    def get_frequent_locations(self, uid, start, end):
+
+        transactions = DiningTransaction.objects.filter(
+            account=uid, date__gte=start, amount__lt=0
+        ).exclude(description__icontains="meal_plan")
+
+        if len(transactions) < 10:
+            return None
+
+        venue_spends = (
+            transactions.values("description")
+            .annotate(total_spend_venue=Sum("amount"))
+            .order_by("total_spend_venue")
+        )
+
+        card = {
+            "type": "frequent-locations",
+            "data": [],
+        }
+
+        venues = [
+            {
+                "location": map_to_pretty(venue["description"]),
+                "week": 0,
+                "month": 0,
+                "semester": venue["total_spend_venue"] * -1,
+            }
+            for venue in venue_spends
+        ][:5]
```

Comment:
why `:5`? is this special in any way? If so, it should probably be factored out into a constant.